### ISSUE TITLE
Format option doesn't work

### ISSUE THREAD

**Issue by sts-ryan-holton** [mariomka/vue-datetime#151]
Latest version of Vue JS and the plugin is at the latest version.

The `format` option doesn't appear to work, this is my setup:

```
<datetime type="date" v-model="formData.testDate" input-class="form-control" format="DateTime.DATE_SHORT"></datetime>
```

My output, here's an image of the input:

![Screenshot 2019-06-11 at 13 49 28](https://user-images.githubusercontent.com/25177538/59273302-c6600f80-8c4f-11e9-939f-4b4e4c0bc12a.png)

**Comment by danielfoxp2** [comment]
Having the same problem. Even the example of the demo is not working.


As a workaround I stay listening to the datetime close event, get the value and create a new vanilla js date passing the string returned by the vue datetime to it. Then I call whatever function I want to format. 

```javascript
const formattedDate = new Date(dateStringProvidedByVueDatetime);
return formattedDate.toLocaleString()
```

**Comment by dnnsldr** [comment]
Use this in your template:
<datetime v-model="the_dob" :type="date" :format="DATE_SHORT"></datetime>

and then in your data											
DATE_SHORT: {
 'DateTime': 'DATE_SHORT'
}

This will use the Luxon formatting: https://moment.github.io/luxon/docs/manual/formatting.html#tolocalestring--strings-for-humans-

**Comment by giacomotontini** [comment]
> Use this in your template:
> and then in your data
> DATE_SHORT: {
> 'DateTime': 'DATE_SHORT'
> }
> 
> This will use the Luxon formatting: https://moment.github.io/luxon/docs/manual/formatting.html#tolocalestring--strings-for-humans-

Could you provide a complete solution?

**Comment by dnnsldr** [comment]
Here is my complete solution. I have three date pickers on a single vue template. There is a datepicker for the v-model "patient_dob", v-model "dos", and v-model "dueDate". The model patient-dob does not have a default but the other two are formatted to use ( 01/01/2019 ) and they both have defaults dates set for them. The code shows how to default the dos model to todays date and the dueDate model to add one month to todays date. 

**FIRST**
in your app.js file make sure to import the necessary files

```
import Datetime from 'vue-datetime';
// plugin loader for CSS files provided by vue datetime
import 'vue-datetime/dist/vue-datetime.css';
// my custom css file for the datepicker
import '../vendor/dateTime.css';
```

**NEXT**
Then in the vue template where you need the date picker I have the follow js. I use DATE_SHORT as a way to format the ISO format that Luxon uses so that it is readable to the end user. Luxon has many preset formats, just reference their documentation. I used the DATE_SHORT , but you could change it to use DATE_MED or DATE_HUGE just as easy. I also set my Locale to English. Notice that in the data() below the dos and the dueDate are setting defaults to ISO. I then convert that ISO to whatever format I need right after that in the DATE_SHORT variable. 

The variable DATE_SHORT could have been anything, the important part is the: 'DateTime': 'DATE_SHORT'

The setZone just calls the function to accurately get the timezone. May not be necessary for your purpose.

```
<script>
// import luxon into the component
import { DateTime as LuxonDateTime } from 'luxon';
import { Settings } from 'luxon';
Settings.defaultLocale = 'en';

// small function to get the timezone accurate
function getCurrentZone(){
	var d = new Date();
	var z = -(d.getTimezoneOffset() / 60);
	return 'UTC'+z;
}

// the export
export default {
	data() {
		return {
                     // For the Date of Service datepicker 
                    // sets the default for the v-model dos to todays date
                     dos: LuxonDateTime.local().toISO(),

                     // for the Due Date datepicker component
                     // sets the default v-model dueDate to todays date + 1 month
		    dueDate: LuxonDateTime.local().plus({ months: 1 }).toISO(),

                    // sets the format
                    DATE_SHORT: {
			'DateTime': 'DATE_SHORT'
		    },

                    // sets the time zone
		    setZone: getCurrentZone(),

                   // make sure you can't have a date in the future
                   // used for Date of Service and Patient DOB
                   maxDateTime: LuxonDateTime.local().toISO()
                 }
	},
}
<script>
```

By setting the variable fo DATE_TIME to 'DateTime': 'DATE_SHORT'  the input for the date picker will have the right format of 11/23/2019 and show up as my default field.
Example image: [https://drive.google.com/file/d/172BXEY2_rGLfiBP3gesrC6YJrfboGH_z/view?usp=sharing](url)

Also in my export default{} for my template I make an Axios POST request where I format the date yet again so that it is a SQL formatted date. I do this by setting the variable value in the POST. First I get the dueDate variable from ISO and then change it with toFormat(). I do this for all the dates.
Example:

```
dueDate: LuxonDateTime.fromISO(this.dueDate).toFormat('yyyy-MM-dd 00:00:00')
```

**LASTLY**
I then add this in the template to show the date pickers

```
<datetime 
	v-model="patient_dob" 
	type="date" 
	:format="DATE_SHORT"
	:max-datetime="maxDateTime"
	:value-zone="setZone"
	class="datetimeStyle">
	<template slot="button-cancel">
		<i class="close icon"></i>
		Cancel
	</template>
	<template slot="button-confirm">
		<i class="check icon"></i>
		Confirm
		</template>
</datetime>


<datetime 
	v-model="dos" 
	type="date" 
	:format="DATE_SHORT"
	:max-datetime="maxDateTime"
	:value-zone="setZone"
	class="datetimeStyle">
	<template slot="button-cancel">
		<i class="close icon"></i>
		Cancel
	</template>
	<template slot="button-confirm">
		<i class="check icon"></i>
		Confirm
	</template>
</datetime>


<datetime 
	v-model="dueDate" 
	type="date" 
	:format="DATE_SHORT"
	:value-zone="setZone"
	class="datetimeStyle">
	<template slot="button-cancel">
		<i class="close icon"></i>
		Cancel
	</template>
	<template slot="button-confirm">
		<i class="check icon"></i>
		Confirm
	</template>
</datetime>
```

Multiple date pickers on one template with defaults, formatting, zones, and a way to send the date in a different format.

**Comment by giacomotontini** [comment]
Thank you so much. I appreciate that

**Comment by jamesofficer** [comment]
Not sure why this was closed? This is definitely still broken.

This is what my input looks like when using the format option: 
![image](https://user-images.githubusercontent.com/5257788/71934055-b44f4b00-31f7-11ea-92f6-cdf37e633b9f.png)

Versions:
"vue": "^2.6.10",
"vue-datetime": "^1.0.0-beta.11",

**Comment by mariomka** [comment]
@yahmez00 `Datetime.DATE_SHORT` and other formats are values provided by luxon.

Check out this fiddle: https://jsfiddle.net/hp35noec/

**Comment by samjarmakani** [comment]
Hi @mariomka, I can assure you this is broken. I've tried every example out there and I get similar output to what @jamesofficer was getting.

I've done exactly what you have done in your fiddle.

First I installed and imported Luxon:
```
const { DateTime } = require("luxon")
```

Defined the format:
```
data() {
   return {
      dateFormat: DateTime.DATETIME_MED
   }
}
```

And then attached it to the component:

```
<datetime v-model="newValue" type="datetime" input-class="form-control" use12-hour :format="dateFormat" value-zone="UTC" zone="UTC"></datetime>
```

I've tried many of the presets, they either do nothing or come out wrong. Usually it results in something weird like adding "UTC" on to the end or removing the time.  I've tried the tokens using `.toFormat` and it usually results in gibberish.
